### PR TITLE
fix(codegen): add proper spacing in anonymous struct compound literals

### LINKED ISSUE #875: Anonymous struct compound literal missing spaces between tokens
## Description

When a struct has an anonymous struct field (e.g., bitfield flags), the transpiler reconstructs the anonymous struct type for compound literal casts but concatenates all tokens without spaces, producing invalid C/C++.

## Minimal Reproduction

**anon.h:**
```c
typedef struct {
    int value;
    struct {
        unsigned int flag_a: 1;
        unsigned int flag_b: 1;
    } flags;
} config_t;
```

**bug.cnx:**
```
#include "anon.h"

void test() {
    config_t cfg <- {
        value: 42,
        flags: { flag_a: true }
    };
}
```

**Run:** `cnext bug.cnx --cpp`

## Actual Output

```cpp
config_t cfg = (config_t){ .value = 42, .flags = (struct{unsignedintflag_a:1;unsignedintflag_b:1;}){ .flag_a = true } };
```

Token concatenation: `unsignedintflag_a` instead of `unsigned int flag_a`.

## Expected Output

```cpp
config_t cfg = (config_t){ .value = 42, .flags = (struct { unsigned int flag_a: 1; unsigned int flag_b: 1; }){ .flag_a = true } };
```

## GCC Output

```
error: 'unsignedintflag_a' does not name a type
error: 'unsignedintflag_b' does not name a type
```

## Workaround

Set anonymous struct fields separately after initialization:
```
config_t cfg <- { value: 42 };
cfg.flags.flag_a <- true;
```

## Version

cnext v0.2.2 (regression — v0.2.1 used the parent type cast which was wrong but didn't have this specific issue)

### PR DESCRIPTION
**Pull request by jlaustill**
## Summary

- Fixed token concatenation bug in anonymous struct compound literal generation
- Added `reconstructAnonymousStruct()` and helper methods to properly format anonymous struct types
- Added unit tests for `DeclaratorUtils.extractTypeFromSpecQualList()`
- Added integration test in `tests/bugs/issue-875-anon-struct-spacing/`

## Test plan

- [x] Unit tests pass (`npm run unit`)
- [x] Integration tests pass (`npm test`)
- [x] All quality checks pass (spelling, linting, type checking)
- [x] Bug reproduction test added and passing

Fixes #875

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by jlaustill — COMMENTED** [review]
## Review Summary

**PR:** #876 — fix(codegen): add proper spacing in anonymous struct compound literals
**Reviewed by:** Claude (automated review)

### Overview
This PR correctly fixes issue #875 where anonymous struct types were being reconstructed with concatenated tokens (e.g., `unsignedintflag_a`) instead of properly spaced output (`unsigned int flag_a`). The solution adds AST reconstruction methods that walk the parse tree and join tokens with appropriate spacing.

### Findings

| Severity | Count |
|----------|-------|
| Critical | 0 |
| Important | 0 |
| Moderate | 2 |
| Nit | 2 |

### Key Issues
- **Moderate**: Potential recursive loop in `extractTypeFromSpecQualList()` → `reconstructAnonymousStruct()` → `extractTypeFromSpecQualList()` needs review
- **Moderate**: Missing test coverage for union types and edge cases

### What Looks Good
- Clean separation of concerns with private helper methods
- Good docstrings explaining each method's purpose
- The bug reproduction test directly exercises the reported issue
- Unit tests properly navigate the AST to test the exact failing scenario

---
🤖 *Automated review by Claude Code*

File: src/transpiler/logic/symbols/c/utils/__tests__/DeclaratorUtils.test.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      const specQualList = flagsDecl?.specifierQualifierList();
+
+      const result = DeclaratorUtils.extractTypeFromSpecQualList(specQualList);
+
+      // Should have proper spacing, not concatenated tokens like "struct{...}"
+      expect(result).toContain("struct {");
+      // Note: our reconstruction adds spaces around the colon, which is valid C
+      expect(result).toContain("unsigned int flag_a : 1;");
+      expect(result).toContain("unsigned int flag_b : 1;");
+    });
```
[the remark this answers is not in the document]
Comment (jlaustill) [reply src/transpiler/logic/symbols/c/utils/__tests__/DeclaratorUtils.test.ts]: Added union test coverage in commit. The test verifies that `union { int as_int; float as_float; }` produces the correct keyword and spacing.
